src/http2_connection.erl

```diff
@@ -1172,27 +1136,36 @@ go_away(ErrorCode,
 %% finds one, it delegates sending the rst_stream frame to it, but if
 %% it doesn't, it seems like a waste to spawn one just to kill it
 %% after sending that frame, so we send it from here.
--spec rst_stream(stream_id(), error_code(), connection()) ->
+-spec rst_stream(stream(),
+                 error_code(),
+                 connection()) ->
                         {next_state, connected, connection()}.
-rst_stream(StreamId, ErrorCode, Conn) ->
-    case get_stream(StreamId, Conn#connection.streams) of
-        false ->
-            RstStream = http2_frame_rst_stream:new(ErrorCode),
-            RstStreamBin = http2_frame:to_binary(
-                             {#frame_header{
-                                 stream_id=StreamId
-                                },
-                              RstStream}),
-            sock:send(Conn#connection.socket, RstStreamBin);
-        Stream ->
-            http2_stream:rst_stream(Stream#stream.pid, ?PROTOCOL_ERROR)
-    end,
+rst_stream(#active_stream{
+              pid=Pid
+             },
+           ErrorCode, Conn)
+  when is_pid(Pid) ->
+    %% http2_stream's rst_stream will take care of letting us know
+    %% this stream is closed and will send us a message to close the
+    %% stream somewhere else
+    http2_stream:rst_stream(Pid, ErrorCode),
+    {next_state, connected, Conn};
+rst_stream(#closed_stream{id=StreamId},
```

Comment:
It's all about the pattern matches of these records, to decide whether or not to try and have the http2_stream fsm handle the rst_stream, or just building a frame at the connection level if there's no running fsm for the stream